lib/internal/Magento/Framework/Mview/View/Changelog.php

```diff
@@ -170,22 +169,19 @@ public function getList($fromVersionId, $toVersionId)
      *
      * @return int
      * @throws ChangelogTableNotExistsException
-     * @throws RuntimeException
      */
     public function getVersion()
     {
         $changelogTableName = $this->resource->getTableName($this->getName());
         if (!$this->connection->isTableExists($changelogTableName)) {
             throw new ChangelogTableNotExistsException(new Phrase("Table %1 does not exist", [$changelogTableName]));
         }
-        $row = $this->connection->fetchRow('SHOW TABLE STATUS LIKE ?', [$changelogTableName]);
-        if (isset($row['Auto_increment'])) {
-            return (int)$row['Auto_increment'] - 1;
-        } else {
-            throw new RuntimeException(
-                new Phrase("Table status for %1 is incorrect. Can`t fetch version id.", [$changelogTableName])
-            );
-        }
+        return (int)$this->connection->fetchOne(
+            $this->connection->select()
```

Comment:
Hi @akaplya,
Looks like in case if this table will be empty (for instance new installation) - we’ll have exception, and it will be regression.

Could you update your code and cover this case with some test?

Reply:
Hii @ihor-sviziev if a table will be empty query will return null, which will be converted to 0.

Reply:
Ah, you’re right 👍